Dockerfile

```diff
@@ -1,53 +1,84 @@
 FROM debian:9-slim AS pim
 MAINTAINER pierre.allard@akeneo.com
 WORKDIR /home/akeneo/pim-docs/
+ENV DEBIAN_FRONTEND=noninteractive
 
-# Sphinx installation
-RUN apt-get update && \
-    apt-get install -y python-pip git
-RUN pip install --upgrade pip
-RUN pip install sphinx~=1.5.3 && \
-    pip install git+https://github.com/fabpot/sphinx-php.git && \
-    pip install git+https://github.com/mickaelandrieu/sphinxcontrib.youtube.git
-RUN apt-get clean && apt-get --yes --quiet autoremove --purge && \
-    rm -rf /var/lib/apt/lists/*
-
-# Copy script
 COPY build.sh /home/akeneo/pim-docs/build.sh
-RUN chmod +x /home/akeneo/pim-docs/build.sh
 
-# Akeneo PIM installation
-ENV DEBIAN_FRONTEND=noninteractive
+# Create environment for build with php, python, mysql and composer
 RUN apt-get update && \
-    apt-get install -y wget unzip
-
-# Install mysql-server
-RUN apt-get install -y lsb-release apt-transport-https ca-certificates && \
+    apt-get install -y --no-install-recommends wget lsb-release apt-transport-https ca-certificates gnupg unzip \
+        python python-setuptools && \
+    #
+    # Add source for mysql
     wget -O /tmp/mysql-apt-config.deb https://dev.mysql.com/get/mysql-apt-config_0.8.7-1_all.deb && \
     dpkg -i /tmp/mysql-apt-config.deb && \
-    apt-get update && \
-    apt-get install -y mysql-server && \
-    rm /tmp/mysql-apt-config.deb
-
-# Install php
-RUN apt-get install -y apt-transport-https ca-certificates && \
+    #
+    # Add source for php
     wget -O /etc/apt/trusted.gpg.d/php.gpg https://packages.sury.org/php/apt.gpg && \
     sh -c 'echo "deb https://packages.sury.org/php/ stretch main" > /etc/apt/sources.list.d/php.list' && \
-    apt update && \
-    apt-get install -y php7.1-apcu php7.1-bcmath php7.1-cli php7.1-curl php7.1-fpm php7.1-gd php7.1-intl php7.1-mcrypt php7.1-mysql php7.1-soap php7.1-xml php7.1-zip php7.1-mbstring && \
+    #
+    # Add sphinx
+    wget -O /tmp/sphinx.zip https://github.com/sphinx-doc/sphinx/archive/v1.5.6.zip && \
+    unzip /tmp/sphinx.zip -d /tmp/ && \
+    cd /tmp/sphinx-1.5.6/ && \
+    python setup.py install && \
+    #
+    # Add youtube-sphinx extension
+    wget -O /tmp/sphinxcontrib.youtube.zip \
+        https://github.com/mickaelandrieu/sphinxcontrib.youtube/archive/master.zip && \
+    unzip /tmp/sphinxcontrib.youtube.zip -d /tmp/ && \
+    cd /tmp/sphinxcontrib.youtube-master/ && \
+    python setup.py install && \
+    #
+    # Add sphinx-php extension
+    wget -O /tmp/sphinx-php.zip \
+        https://github.com/fabpot/sphinx-php/archive/v1.0.10.zip && \
+    unzip /tmp/sphinx-php.zip -d /tmp/ && \
+    cd /tmp/sphinx-php-1.0.10/ && \
+    python setup.py install && \
+    #
+    # Download packages
+    apt-get update && \
+    apt-get install -y --no-install-recommends mysql-server php7.1-apcu php7.1-bcmath \
+        php7.1-cli php7.1-curl php7.1-fpm php7.1-gd php7.1-intl php7.1-mcrypt php7.1-mysql php7.1-soap php7.1-xml \
```

Comment:
PHP 7.1 for a PIM 1.x? Shouldn't you use 5.6?

Reply:
We can use 7.1 or 5.6 for 1.7. I put 7.1 to avoid multiple layers through upper versions

Reply:
OK then.